proposals/Imperative-Shadow-DOM-Distribution-API.md

```diff
@@ -54,36 +76,55 @@ partial interface HTMLSlotElement {
 
 `slot.assign(sequence<Node> nodes)` runs the following steps:
 
-1. Set the slot's *manually-assigned-nodes* to *nodes*.
-2. Run [assign slotables for a tree] with slot’s tree.
+1. Check `slotAssignment` is "manual", throw otherwise.
+2. Validate each node in nodes that its node.parentNode == slot.rootNode.host. Throw if not equal.
+3. Set the slot's `manually-assigned-nodes` to `nodes`.
+4. Run [assign slotables for a tree] with slot’s tree.
 
-step 2 is required because we have to re-calculate [assigned nodes] of every slots in the tree at this timing.
+Step 4 is required because we have to re-calculate [assigned nodes] of every slot in the tree at this timing.
 
-Note: The detail is explained later, however, it would be worth noting that *manually-assigned-nodes* is not used as
-[assigned nodes] as is. You can think that `slot.assign(sequence<Node> nodes)` tell the engine *candidate nodes* from where [assigned nodes] are constructed.
+Note: The detail is explained later, however, it would be worth noting that `manually-assigned-nodes` is not used as
+[assigned nodes] as is. You can think of `slot.assign(sequence<Node> nodes)` as telling the engine a set of "candidate nodes" from which [assigned nodes] are constructed.
 
 [assign slotables for a tree]: https://dom.spec.whatwg.org/#assign-slotables-for-a-tree
 
-# Changes to DOM Standard
+## Changes to DOM Standard
 
-## ShadowRootInit
+### ShadowRootInit
 
 
 ``` webidl
 ShadowRootInit {
```

Comment:
It would be good to rewrite this closer to valid Web IDL.

Reply:
Updated with WebIDL